api_app/analyzers_manager/observable_analyzers/spyse.py

```diff
@@ -0,0 +1,57 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import requests
+import re
+from api_app.exceptions import AnalyzerRunException, AnalyzerConfigurationException
+from api_app.analyzers_manager import classes
+
+from tests.mock_utils import if_mock_connections, patch, MockResponse
+
+
+class Spyse(classes.ObservableAnalyzer):
+    base_url: str = "https://api.spyse.com/v4/data/"
+
+    def set_params(self, params):
+        self.analysis_type = params.get("spyse_analysis", "search")
+        self.__api_key = self._secrets["api_key_name"]
+
+    def run(self):
+        if self.analysis_type == "search":
+            params = {"key": self.__api_key, "minify": True}
+
+
+        if self.observable_classification == self.ObservableTypes.DOMAIN:
+            uri = f"domain/{self.observable_name}"
+        elif self.observable_classification == self.ObservableTypes.IP:
+            uri = f"ip/{self.observable_name}"
+        elif self.observable_classification == self.ObservableTypes.GENERIC:
+            if re.match(r"^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$", self.observable_name):
+                uri = f"email/{self.observable_name}"
+        else:
+            raise AnalyzerRunException(
+                f"not supported observable type {self.observable_classification}. "
+                "Supported are IP, Domain and Generic"
```

Comment:
I think this is redundant as the analyzer would run only for these 3 types, as configured by you in `analyzer_config.json`

Reply:
Please run `black` and `flake8` on the python file, which will fix the LGTM's warnings as well.

https://intelowl.readthedocs.io/en/latest/Contribute.html#code-style

Reply:
@Palash-Vishnani
for formatting, I would suggest setting up `pre-commit` (https://pre-commit.com/). It will automatically run `flake8` and `black` before each commit so you won't have to worry about formatting issues.